### PR TITLE
mv tupl to conceptual-tupl

### PR DESCRIPTION
**Pull request by xfthhxk**

### CODE REVIEW

File: src/clj/conceptual/core.clj [hunk 1]
```diff
@@ -433,54 +433,61 @@
...
-;; TODO: support more compaction target types
+
+(defn- db-type
+  [db]
+  (let [db-type (type db)]
+    (if (instance? Class db-type)
+      (.getName ^Class db-type)
+      db-type)))
+
+(defmulti compact-db! (fn [db _type] [(db-type db) _type]))
```
Comment (xfthhxk) [inline src/clj/conceptual/core.clj:444]: wonder if there should be a java interface for compact, pickle, and unpickle?
With the multimethod approach would need a defmethod for each subtype of a DB class.  Protocol would be another option.

File: src/clj/conceptual/word_fixes.clj [hunk 2]
```diff
@@ -52,114 +50,82 @@
...
+  and returns an integer set."
+  [ids-by-infix-fn s]
   (->> (keyword->words s)
-       (map ids-by-infix)
-       (filter (comp not nil?))
+       (map ids-by-infix-fn)
+       (filter some?)
        (apply int-sets/intersection)))
 
-(defn by-prefix-phrase [s]
```
Comment (xfthhxk) [inline src/clj/conceptual/word_fixes.clj:141]: these fns don't seem to buy much

File: src/clj/conceptual/word_fixes.clj [hunk 3]
```diff
@@ -52,114 +50,82 @@
...
-(comment (.isClosed (tupl/index :name->id)))
-
-
-(defn ids-by-prefix [s]
-  (tupl/load :prefix s))
-
-(defn ids-by-infix [s]
-  (tupl/load :infix s))
-
-(defn by-prefix [s]
```
Comment (xfthhxk) [inline src/clj/conceptual/word_fixes.clj:113]: these fns don't seem to buy much

File: src/clj/conceptual/core.clj (hunk 1 above)
Comment (jekyllislandtours) [reply src/clj/conceptual/core.clj:444]: multimethod would be better I feel in this case

File: src/clj/conceptual/word_fixes.clj (hunk 3 above)
Comment (jekyllislandtours) [reply src/clj/conceptual/word_fixes.clj:113]: sure

**Review by jekyllislandtours — APPROVED** [review]

**Pull request merged**